Speedup lighting shader parameters

Xaxetrov opened the pull request:
Needed to use this commit to its full potential :
243d73e893998147206a4017df6fd327d0306ae3 (gitlab)

martinken committed:
add centralized lighting uniforms and benchmark

profiling has shown that a lot of time is spent in the
mapper setting the lighting uniforms when there are a
large number of mappers. In a more traidiotnal program
this is handled by a UBO or similar set once per program.
This topic addds in capability to the shader program
to store an mtime associated with a group of uniforms.
It also makes use of that capability by having the renderer
generate shader code for the lights, and update the unifoms
when requested and needed. The result is a significant reduction
in the time spect updating the lights when faced with many mappers.

This topic also includes a new benchmark to test the performance
of many actors.

This topic also includes a performance fix to short circuit
the mtime check for RenderPasses in the mapper. The most common
case is none and yet that case was taking significant time.

martinken committed:
add missing modifieds when changing shader replacements

Modifieds were missing causing shaders to not be rebuilt
when they should have been

(cherry picked from commit d592cb2a44b93290c48fb1df3ed6d25a7e534eb6)

finetjul commented on Rendering/OpenGL2/vtkOpenGLRenderer.cxx:
```diff
@@ -997,3 +1069,116 @@ int vtkOpenGLRenderer::GetPickedIds(unsigned int atMost,
   return k;
 }
+
+}
```
this seems odd, are you sure you are not missing another commit ?

Xaxetrov replied:
Damn I had correct this automerge on the old PR

Commit:
Fix file unconsistency due to cherry-picking

The last two commits are cherry-picks that introduce a small error if not
applied on top of some older commits. This one tempoarly fix this until
we upgrade to a newer clean VTK.

Xaxetrov commented:
@finetjul Temporary fix works on my computer. Check if all is ok with you.
